lineDrawer.py

```diff
@@ -32,13 +33,21 @@ def mouseMoveEvent(self, event):
     def mouseReleaseEvent(self, event):
         if config.draw_wire_state:
             self.line_done = 1
-            config.lines.append((self.pos1, self.pos2))
+            ret = self.setObjectConnection()
+            if ret >= 0:
+                config.lines.append((self.pos1, self.pos2))
+                print(config.current_sym_object.connections)
+            else:
+                ok = QMessageBox.about(self, "Alert", "Invalid line")
+                if not ok:
+                    pass
             self.pos1 = None
             self.pos2 = None
 
+
     def paintEvent(self, event):
```

Comment:
Might want to add a condition to check the draw wire state before calling `self.update()` since it causes laggy behavior otherwise.